Lib/dbm/sqlite3.py

```diff
@@ -0,0 +1,117 @@
+import os
+import sqlite3
+from contextlib import suppress, closing
+from collections.abc import MutableMapping
+
+BUILD_TABLE = """
+  CREATE TABLE IF NOT EXISTS Dict (
+    key TEXT NOT NULL,
+    value BLOB NOT NULL,
+    PRIMARY KEY (key)
+  )
+"""
+GET_SIZE = "SELECT COUNT (key) FROM Dict"
+LOOKUP_KEY = "SELECT value FROM Dict WHERE key = ? LIMIT 1"
+STORE_KV = "REPLACE INTO Dict (key, value) VALUES (?, ?)"
+DELETE_KEY = "DELETE FROM Dict WHERE key = ?"
+ITER_KEYS = "SELECT key FROM Dict"
+
+
+class error(OSError):
+    pass
+
+
+def _normalize_uri_path(path):
+    path = path.replace("?", "%3f")
+    path = path.replace("#", "%23")
+    while "//" in path:
+        path = path.replace("//", "/")
+    return path
+
+
+class _Database(MutableMapping):
+
+    def __init__(self, path, flag):
+        match flag:
+            case "r":
+                flag = "ro"
+            case "w":
+                flag = "rw"
+            case "c":
+                flag = "rwc"
+            case "n":
+                flag = "rwc"
+                try:
+                    os.remove(path)
+                except FileNotFoundError:
+                    pass
+            case _:
+                raise ValueError("Flag must be one of 'r', 'w', 'c', or 'n', not",
+                                 repr(flag))
+
+        # We use the URI format when opening the database.
+        path = os.fsdecode(path)
+        if not path or path == ":memory:":
+            uri = "file:?mode=memory"
```

Comment:
It should be documented.

Reply:
Yes, it should; thanks!

Reply:
Perhaps we should not allow it; this is not sqlite3, it is dbm. I suggest we remove this "feature" and treat everything as an ordinary filename.

Reply:
There is enough time till the next alpha to decide if we want this or not; let's leave it out for now, at least.

Reply:
Also, does it work for filenames like "localhost" or "/localhost"?

Reply:
> Also, does it work for filenames like "localhost" or "/localhost"?

Those are not special names; they will be accepted as ordinary file names.

Reply:
Let me take that back; `"localhost"` is considered special by SQLite. Let's add more tests.